src/scraping/start_for_life/scrape_startforlife.py

```diff
@@ -0,0 +1,187 @@
+"""
+Get content from all URLs from the NHS https://www.nhs.uk/start-for-life/site-map/
+
+Usage:
+
+First navigate to the scraping folder
+$ cd src/scraping/start_for_life
+
+Then run the script
+$ poetry run python scrape_startforlife.py
+
+The final output file is stored in data/startforlife.csv
+with the following columns:
+    - URL: URL of the page
+    - content_type: Content type (ie, class tag of the div section)
+    - header: Header of the page's section
+    - content: Content of the page's section
+    - content_no: Index of the section (ie, if there are multiple sections on the page, this is the integer index of the section)
+
+"""
+
+import logging
+
+from time import sleep
+from typing import List
+
+import pandas as pd
+import requests
+
+from bs4 import BeautifulSoup
+from tqdm import tqdm
+
+
+# URLs to scrape
+SITEMAP_PATH = "data/sitemap.csv"
+BASE_URL = "https://www.nhs.uk/"
+# Headers to be used when scraping
+REQUEST_HEADERS = {
+    "User-Agent": "Data collection for the purpose of research. For questions, reach out to karlis.kanders@nesta.org.uk"
+}
+# We're only fetching on type of div section; one can also add 'nhsuk-promo__content' to capture the site navigation content
+DIV_CLASS = ["nhsuk-u-reading-width"]
+# Typical header tags
+HEADER_TAGS = ["h1", "h2", "h3", "h4", "h5", "h6"]
+# Sections with these header will be excluded from the final output
+EXCLUDED_HEADERS = ["Sign up for emails"]
+# Path to the output file
+OUTPUT_PATH = "data/start_for_life.csv"
+
+
+def merge_sections_based_on_headers(headers: List[str], sections: List[str]) -> (List[str], List[str], List[int]):
+    """
+    Merge sections based on headers, where sections without headers will be merged in the previous section that has a header
+
+    Args:
+        headers
+            List of headers
+
+        sections
+            List of sections
+
+    Returns:
+        Tuple of lists with headers, merged sections and removed indices
+    """
+    new_headers = []
+    new_sections = []
+    # Keep track of removed indices
+    removed_indices = []
+    # Keep track of the last non-empty header index
+    last_non_empty_index = -1
+
+    for i in range(len(headers)):
+        # If header is non-empty
+        if headers[i]:
+            new_headers.append(headers[i])
+            new_sections.append(sections[i])
+            last_non_empty_index = len(new_headers) - 1
+        # If header is empty
+        else:
+            # Add index to removed list
+            removed_indices.append(i)
+            # If there was a previous non-empty header
+            if last_non_empty_index != -1:
+                new_sections[last_non_empty_index] += sections[i]
```

Comment:
This is a totally valid style but I am a huge fan of [asking for forgiveness instead of permission](https://stackoverflow.com/questions/12265451/ask-forgiveness-not-permission-explain). It's considered more pythonic too.

The alternative would be something like:

```python
try:
    for header in headers:
        ...
except IndexError:
    ...
```